### PR TITLE
feat: Allow passing repo config path via flag

### PR DESCRIPTION
**Pull request by achals**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: docs/tutorials/structuring-feature-repos.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# How to Structure Feature Repositories
```
Comment (felixwang9817) [inline docs/tutorials/structuring-feature-repos.md:1]: belongs in another PR?

File: sdk/python/feast/repo_operations.py [hunk 2]
```diff
@@ -325,13 +325,12 @@ def registry_dump(repo_config: RepoConfig, repo_path: Path) -> str:
     return json.dumps(registry_dict, indent=2, sort_keys=True)
 
 
-def cli_check_repo(repo_path: Path):
+def cli_check_repo(repo_path: Path, fs_yaml_file: Path):
```
Comment (felixwang9817) [inline sdk/python/feast/repo_operations.py:328]: nitty but it's slightly confusing to me that `cli_check_repo` and `load_repo_config` have the args switched in order; is it possible to use the same order?

File: sdk/python/feast/cli.py [hunk 3]
```diff
@@ -72,8 +72,17 @@ def format_options(self, ctx: click.Context, formatter: click.HelpFormatter):
     default="info",
     help="The logging level. One of DEBUG, INFO, WARNING, ERROR, and CRITICAL (case-insensitive).",
 )
+@click.option(
```
Comment (felixwang9817) [inline sdk/python/feast/cli.py:75]: can you add a test for this option? maybe in one of the CLI tests?

File: sdk/python/feast/feature_store.py [hunk 4]
```diff
@@ -114,30 +114,39 @@ class FeatureStore:
...
         """
         Creates a FeatureStore object.
 
         Raises:
             ValueError: If both or neither of repo_path and config are specified.
         """
-        if repo_path is not None and config is not None:
-            raise ValueError("You cannot specify both repo_path and config.")
-        if config is not None:
+        if fs_yaml_file is not None and config is not None:
```
Comment (felixwang9817) [inline sdk/python/feast/feature_store.py:132]: hmm shouldn't this continue to check `repo_path`? as well as the logic below? since we still need exactly one of `repo_path` and `config`, and if `repo_path` is specified, there's an option for `fs_yaml_file` to be specified as well

**Review by felixwang9817 — COMMENTED** [review]
looks good overall!

File: sdk/python/feast/repo_operations.py (hunk 2 above)
Comment (achals) [reply sdk/python/feast/repo_operations.py:328]: sure makes sense

File: sdk/python/feast/feature_store.py (hunk 4 above)
Comment (achals) [reply sdk/python/feast/feature_store.py:132]: I think we actually need repo path regardless, and one of config or fs_yaml_path.

... I may change the FS constructor cause I feel like we should be able to only pass in in-memory objects to construct it. Dealing with the file system etc should not be something the FS has to care about (ideally)

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Comment by adchia** [comment]
@achals can you also add docstrings for this?

**Comment by cburroughs** [comment]
Does this resolve #1556?

**Comment by achals** [comment]
> Does this resolve #1556?

Yes!

File: sdk/python/feast/feature_store.py [hunk 5]
```diff
@@ -1569,7 +1581,7 @@ def _get_online_features(
         }
 
         # If the embedded Go code is enabled, send request to it instead of going through regular Python logic.
-        if self.config.go_feature_retrieval:
+        if self.config.go_feature_retrieval and self._go_server:
```
Comment (idahoakl) [inline sdk/python/feast/feature_store.py:1584]: @achals What was the reason for adding `and self._go_server` to this condition expression?  This seems to be the cause of why the go server will not start and trying to understand if there was another issue that was being solved for.
